Oversettelse development/data og development/ux til engelsk (#594)
Altinn/altinn-studio-docs: content/app/development/logic/validation/_index.en.md

| required  | 'Du må fylle ut {0}'          | 'Du må fylle ut {0}'          | 'You have to fill out {0}'            |
| enum      | 'Kun verdiene {0} er tillatt' | 'Kun verdiene {0} er tillatt' | 'Only the values {0} are permitted'   |

> ### Specifically on default error messages for required fields

Reviewer: ```suggestion
### More about error messages for required fields
```
This part also had an existing English translation, btw

The error message for required fields is as defined above, _"You have to fill out {0}"_. The `{0}` symbol is replaced with the field that
the error message is shown for. This is done in the following way:

> - Uses the field's `shortName` text. This is a new component, that can be set up on a component in the same way as a prompt (`title`).  _This is currently used only for an error message for required fields._

Reviewer: ```suggestion
- If `shortName` text is defined for the component, this is used. _This is a new text that is currently used only for this specific error message._
```
I preferred the original text here

{{% notice warning %}}
Note that when the XSD is changed, the custom error messages will de removed from the JSON schema.
In the future, there will be support for setting custom error messages in the data modelling tool in Altinn Studio. But for now, this is a manual job.
{{% /notice %}}

## Server-side validation

Server-side validation can be split into two categories:

> - **Validations against data model** - These are run automatically whenever the user attempts to send inn form data.
> - **Custom validations** - these are written by the application developer,
> and are run when the user attempts to submit form data or move the process to a new step.

Reviewer: ```suggestion
- **Validations against data model** - These run automatically whenever the user attempts to submit form data.
- **Custom validations** - these are written by the application developer,
and run when the user attempts to submit form data or move the process to a new step.
```

}
```

See comments in code above for an expnation of what the different parts do.

> In the other parameter of the method `AddModelError`, where it says "_Error: First name cannot contain the value '1337'_", you can use a text key for a [text defined in the resource files](../../ux/texts) for language support. 

Reviewer: ```suggestion
In the other parameter of the method `AddModelError`, where it says "_Error: First name cannot contain the value '1337'_", you can use a text key for a [text defined in the resource files](../../ux/texts) for multilingual support.
```

```

{{% notice warning %}}

> Note that if you define a field to trigger validation on server-side, only the result of this validation will be displayed. E.i.
> if there is another client-side validation defined, a possible server-side validation of the field will overrun these. Therefore, you should make sure

Reviewer: ```suggestion
Note that if you define a field to trigger validation server-side, only the result of this validation will be displayed. Meaning,
if there is another client-side validation defined, a possible server-side validation of the field will overwrite these. Therefore, you should make sure
```

    {
        validationResults.AddModelError(
            nameof(model.boaddresse), 
            "Address can not be longer than 150 characters.");
    }
}
```

### Specify that validation errors are fixed
When validation is triggered by a single field, all former validations on this field will be removed pending a response from the last validation.

> If a field triggers validation that updates/adds an error message to multiple fields at once, these will not be removed even though there is no longer any
> error in these fields. This is because there is no way to know which fields may have been validated through a single field validation.

Reviewer: ```suggestion
If a field triggers validation that updates/adds an error message to multiple fields at once, these will not be removed even when there no longer are any
errors in these fields. This is because there is no way to know which fields may have been validated through a single field validation.
```

}
```

Examples on display of different validations below:

!["Information message"](info-message.jpeg "Example on information message (*INFO* - prefix)" )

!["Success message"](success-message.jpeg "Example on success message (*SUCCESS* - prefix)")

!["Warning message"](warning-message.jpeg "Example on warning message (*WARNING* - prefix)" )

It is also possible to overrule the title you see on the messages by adding the keys `soft_validation.info_title`, `soft_validation.warning_title`, and `soft_validation.success_title` in the text resources if you want to set a custom title.

## Group validation

> It is possible to apply validations to a repeating group when the user wants to set a custom title.

Reviewer: ```suggestion
It is possible to apply validations to a repeating group when the user saves a row in the group.
```
